mlflow/pytorch/__init__.py

```diff
@@ -522,6 +522,9 @@ class LinearNNModel(torch.nn.Module):
     with open(os.path.join(path, conda_env_subpath), "w") as f:
         yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
 
+    pip_req_subpath = "additional_requirements.txt"
```

Comment:
why this name different with other flavors ?

Reply:
It seems that for only pytorch, there's an option to pass in a requirements_file to save_model, which was causing some tests to fail that were checking for the passed requirements file vs the one generated from the conda env. I could alternatively use a different path for that test or change all of the pip_req_subpath to "additional_requirements.txt." Not sure which option is best or if there is some other solution.

Reply:
The right thing to do here is probably to merge these dependencies together, since the name `additional_requirements` is kind of nonstandard. Is this feasible to do? If so, we should consider doing this in a separate PR. If not, we can leave it like this in this PR

Reply:
Resolution offline was to move the Pytorch stuff out into its own smaller PR - there we'll implement the overriding behavior

Reply:
@stevenchen-db @ankit-db The purpose of `requirements_file` in `mlflow.pytorch.{save_model, log_model}` is to make it easier to serve a saved model with [`mlflow-torchserve`](https://github.com/mlflow/mlflow-torchserve) (which serves a a model with [`torchserve`](https://github.com/pytorch/serve)). It works like this:

1. Log a pytorch model with `requirements_file`
2. Serve the model with `mlflow-torchserve`
3. `mlflow-torchserve` reads the logged `requirements_file` and pass it to `torchserve`
4. `torchserve` creates an environment based on the `requirements_file`